Mathlib/LinearAlgebra/UnitaryGroup.lean

```diff
@@ -226,4 +243,28 @@ theorem mem_orthogonalGroup_iff' {A : Matrix n n β} :
 
 end OrthogonalGroup
 
+section specialOrthogonalGroup
+
+variable (n) (β : Type v) [CommRing β]
+
+attribute [local instance] starRingOfComm
+
+/-- `Matrix.specialOrthogonalGroup n` is the group of orthogonal `n` by `n` where the determinant
+is one. (This definition is only correct if 2 is invertible.)-/
+abbrev specialOrthogonalGroup := specialUnitaryGroup n β
+
+variable {n} {β} {A : Matrix n n β}
+
+theorem mem_specialOrthogonalGroup_iff :
+    A ∈ specialOrthogonalGroup n β ↔ A * star A = 1 ∧ A.det = 1 :=
+  mem_specialUnitaryGroup_iff
+
+/-- Given an orthogonal Matrix with the determinant 1, we get an instance of
+`specialOrthogonalGroup n β` -/
+def specialOrthogonalGroup.mkOfDetEqOne (h1 : star A * A = 1) (h2 : A.det = 1) :
+    specialOrthogonalGroup n β :=
```

Comment:
I don't entirely see the point of this `def`, especially not without `@[simps]`; I think it's only going to erect barriers. But more importantly, it again suffers from the problem of choosing sides (i.e., `star A * A = 1` versus `A * star A = 1`). And the version that just assumes `A ∈ orthogonalGroup n β` is truly unnecessary.

Reply:
Ok thank you for the remarks, I'm just gonna remove it then